### PR TITLE
Interpolation Behaviour Message Format

### PR DESCRIPTION
**Pull request by maennchen**
Implements #294

Merge #296 first

This is only a draft and not a final implementation.

It looks like it is expected to configure the extract via the `mix.exs` options and not via the backend. Why is that?

If we actually went this way, it would probably make more sense to yield the format when extracting. This way the extractor itself would not need to know about what format to use.

### CODE REVIEW

**Comment by josevalim** [comment]
Hi @maennchen, thanks for the PR!

I believe the changes could be simpler. Change `flags: MapSet.new([extracted_translations_flag])` to be the flag we got from the interpolator and call it a day? :)

**Comment by maennchen** [comment]
@josevalim Wouldn't that mean, that no messages would be purged if extracting with another interpolator?

(Calling `PO.Translations.autogenerated?`)

Is it ok to use the backend config in extraction? If yes: why are there configs via the `mix.exs` and not only via the backend module?

**Comment by josevalim** [comment]
The configs in mix.exs are because we don't know what are the backends until that moment, iirc.

**Comment by josevalim** [comment]
@maennchen should we use another flag for auto generated? What does gettext use?

**Comment by maennchen** [comment]
@josevalim I'm not aware if gettext itself does something like that. It seems like there's no (documented) flag for that: https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html#index-msgstr

**Comment by maennchen** [comment]
But I think we can just make one up as long as we prefix it. (To make sure it doesn't collide)

Something like `elixir-generated`.

If you would want to go down that path, I would also offer a PR and do that before this PR.

**Comment by josevalim** [comment]
Let's go with elixir-generated, yeah. :+1:

File: lib/gettext/po/translations.ex [hunk 1]
```diff
@@ -14,7 +14,7 @@ defmodule Gettext.PO.Translations do
   @doc """
   Tells whether a translation was manually entered or generated by Gettext.
 
-  As of now, a translation is considered autogenerated if it has the "elixir-format" flag.
+  As of now, a translation is considered autogenerated if it has the "elixir-autogenerated" flag.
```
Comment (maennchen) [inline lib/gettext/po/translations.ex:17]: TODO: CHange when #296 is decided & merged.

**Comment by josevalim** [comment]
Can you please rebase? And then I think this should be good!

**Comment by maennchen** [comment]
@josevalim Done

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**